### PR TITLE
Character after battle#69

### PR DESCRIPTION
**Pull request by NoahBartell**
Resets enemy and reloads player on loss
saves player stats on win
I noticed there might be an issue with CharacterFactory's LoadCharacter method so I might tweak that in this pr if I get to it

### CODE REVIEW

File: dunwoody_disaster/CharacterFactory.py [hunk 1]
```diff
@@ -119,6 +122,15 @@ def reset(self):
...
         self.set_stamina(self.maxStamina)
 
+    def reload(self):
+        """Reloads the character's health, stamina, and magic."""
+        char = CharacterFactory.LoadCharacter(self.name)
+        self = char
+        self.set_health(char.curHealth)
+        self.set_magic(char.curMagic)
+        self.set_stamina(char.curStamina)
+        print(f"loading{self.name} health: {self.curHealth}")
```
Comment (chanson02) [inline dunwoody_disaster/CharacterFactory.py:132]:
```suggestion
```

Comment (chanson02) [inline dunwoody_disaster/CharacterFactory.py]: I think `self` here is a local variable, so any changes made to it won't be affected outside of this scope.
```suggestion
    def reload(self):
        """Reloads the character's health, stamina, and magic from the last save point."""
        char = CharacterFactory.LoadCharacter(self.name)
        self.set_health(char.curHealth)
        self.set_magic(char.curMagic)
        self.set_stamina(char.curStamina)
```

File: main.py [hunk 2]
```diff
@@ -97,7 +107,7 @@ def loot_collected():
     def showDefeatScreen(self):
         if self.fight is None:
             raise Exception("Defeat Screen expects a fight")
-
+        print("Defeat Screen")
```
Comment (chanson02) [inline main.py]:
```suggestion
```

File: main.py [hunk 3]
```diff
@@ -66,11 +66,20 @@ def showSelector(self):
...
+    def saveCharacter(self, character: Character):
+        CharacterFactory.SaveCharacter(character)
+        print(f"Saved {character.name}")
+
+    def loadCharacter(self, name: str) -> Character:
+        print(f"Loading {name}")
+        return CharacterFactory.LoadCharacter(name)
+
     def createPlayableCharacters(self) -> list[Character]:
         # cooper = CharacterFactory.Cooper()
```
Comment (chanson02) [inline main.py]:
```suggestion
```
I know this wasn't you.

**Review by chanson02 — CHANGES_REQUESTED** [review]
Did you open this at 1 AM? I respect the grind.

File: main.py (hunk 3 above)
Comment (chanson02) [reply main.py]: Looks like I missed that print on line 80.

File: dunwoody_disaster/CharacterFactory.py (hunk 1 above)
Comment (NoahBartell) [reply dunwoody_disaster/CharacterFactory.py]: Yeah i think that was part of the issues I was having

**Review by instMeyFTW — APPROVED** [review]
Nice work!

File: main.py [hunk 4]
```diff
@@ -66,13 +66,21 @@ def showSelector(self):
...
         self.player = character
+        self.saveCharacter(character)
         self.mapScreen = MapScreen(Map.buildMap(self.player))
         self.mapScreen.onEnter(self.EnterFight)
         self.stack.addWidget(self.mapScreen)
         self.showMapScreen()
 
+    def saveCharacter(self, character: Character):
+        CharacterFactory.SaveCharacter(character)
+        print(f"Saved {character.name}")
```
Comment (chanson02) [inline main.py]:
```suggestion
```

File: main.py [hunk 5]
```diff
@@ -66,13 +66,21 @@ def showSelector(self):
...
+
+    def loadCharacter(self, name: str) -> Character:
+        print(f"Loading {name}")
```
Comment (chanson02) [inline main.py]:
```suggestion
```

**Review by chanson02 — APPROVED** [review]
Good work.

**Pull request merged**